### PR TITLE
yle-dl 20210808

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

File: Formula/yle-dl.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class YleDl < Formula
   desc "Download Yle videos from the command-line"
   homepage "https://aajanki.github.io/yle-dl/index-en.html"
-  url "https://github.com/aajanki/yle-dl/archive/20210704.tar.gz"
-  sha256 "775b6a6650d684f63cee1326e1952fcfa02a683ea656fde00d3ffa8038e8af06"
+  url "https://github.com/aajanki/yle-dl/archive/20210808.tar.gz"
```
Comment (cho-m) [inline Formula/yle-dl.rb:4]: Have we ever tried using PyPI URL?
https://files.pythonhosted.org/packages/fa/bb/3e89fa9214d97d2d590fbde45c1ad9e9975aa5dc69a86f7d2d8d3fa20c85/yle-dl-20210808.tar.gz

**Comment by cho-m** [comment]
Test is passing, so only need to update resources, e.g.
```console
❯ pipgrip yle-dl
yle-dl==20210808
attrs==21.2.0
configargparse==1.5.2
lxml==4.6.3
requests==2.26.0
certifi==2021.5.30
charset-normalizer==2.0.4
idna==3.2
urllib3==1.26.6
```

One way to do this would be:
1. make `AdobeHDS.php` the 1st resource
2. comment it out
3. run `brew update-python-resources yle-dl`
4. uncomment `AdobeHDS.php`

This will probably delete the old setuptools, but I don't know if we still need that outdated package.

**Comment by cho-m** [comment]
May also be able to rewrite the entire installation to use virtualenv instead, i.e.
```rb
  def install
    # Fix "ld: file not found: /usr/lib/system/libsystem_darwin.dylib" for lxml
    ENV["SDKROOT"] = MacOS.sdk_path if MacOS.version == :sierra

    venv = virtualenv_create(libexec, "python3")
    venv.pip_install (resources - [resource("AdobeHDS.php")])
    venv.pip_install_and_link buildpath
    resource("AdobeHDS.php").stage(pkgshare)
  end
```

File: Formula/yle-dl.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/yle-dl.rb]: yeah, I dont think we tried it before, IIRC this might be the upstream did not publish artifact to pypi before?

**Review by bayandin — APPROVED** [review]

**Pull request closed without merging**